File: src/main/java/com/nuutrai/reactor/entity/impl/vent/AdvancedVentEntity.java
package com.nuutrai.reactor.entity.impl.vent;

import com.nuutrai.reactor.entity.Sellable;
import com.nuutrai.reactor.entity.lang.VentEntity;
import com.nuutrai.reactor.util.MultiTypeMap;
import org.bukkit.Material;

public class AdvancedVentEntity extends VentEntity {

    public AdvancedVentEntity() {
        super("advanced_vent", Material.IRON_BLOCK);
    }

    @Override
    public void tick(Sellable[] neighbours, MultiTypeMap params) {

    }

    @Override
    public double getSellAmount() {
        return 0;
    }

    @Override
    public void sell() {

    }

    @Override
    public Sellable clone() {
        return new AdvancedVentEntity();
    }
}

File: src/main/kotlin/com/nuutrai/reactor/entity/impl/vent/AdvancedVentEntity.kt
package com.nuutrai.reactor.entity.impl.vent

import com.nuutrai.reactor.entity.Sellable
import com.nuutrai.reactor.entity.lang.VentEntity
import com.nuutrai.reactor.util.MultiTypeMap
import org.bukkit.Material

class AdvancedVentEntity : VentEntity("advanced_vent", Material.IRON_BLOCK) {
	override fun tick(neighbours: Array<Sellable?>, params: MultiTypeMap) {
	}

	override val sellAmount: Double
		get() = 0.0

	override fun sell() {
	}

	override fun clone(): Sellable {
		return AdvancedVentEntity()
	}
}
